### PR TITLE
Document now unecessary workaround for catalogs in KTS `plugins {}` in upgrade guide

### PR DESCRIPTION
**Pull request by eskatos**
This is a follow up to
* https://github.com/gradle/gradle/issues/22797
* https://github.com/gradle/gradle/pull/23639

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc [hunk 1]
```diff
@@ -242,3 +242,16 @@ These members will be removed in Gradle 9.0.
...
+=== Changes in the IDE integration
+
+[[kotlin_dsl_plugins_catalogs_workaround]]
+==== Workaround for false positive errors shown in Kotlin DSL `plugins {}` block using version catalog is not needed anymore
+
+Version catalog accessors for plugin aliases in the `plugins {}` block aren't shown as errors in IntelliJ IDEA and Android Studio Kotlin script editor anymore.
+
+If you were using the `@Suppress("DSL_SCOPE_VIOLATION")` annotation as a workaround, you can now remove it.
+
+If you were using the link:https://plugins.jetbrains.com/plugin/18949-gradle-libs-error-suppressor[Gradle Libs Error Suppressor] IntelliJ IDEA plugin, you can now remove it.
```
Comment (TWiStErRob) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc:255]:
```suggestion
If you were using the link:https://plugins.jetbrains.com/plugin/18949-gradle-libs-error-suppressor[Gradle Libs Error Suppressor] IntelliJ IDEA plugin, you can now uninstall it.
```

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc [hunk 2]
```diff
@@ -242,3 +242,16 @@ These members will be removed in Gradle 9.0.
...
 Please replace it by its equivalent `JvmVendorSpec.IBM` to avoid warnings and potential errors in the next major version release.
+
+=== Changes in the IDE integration
+
+[[kotlin_dsl_plugins_catalogs_workaround]]
+==== Workaround for false positive errors shown in Kotlin DSL `plugins {}` block using version catalog is not needed anymore
+
+Version catalog accessors for plugin aliases in the `plugins {}` block aren't shown as errors in IntelliJ IDEA and Android Studio Kotlin script editor anymore.
+
+If you were using the `@Suppress("DSL_SCOPE_VIOLATION")` annotation as a workaround, you can now remove it.
```
Comment (TWiStErRob) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc:253]: does clear IDE caches need mentioning or was that suggestion more of a precaution?

**Review by TWiStErRob — APPROVED** [review]

**Review by hythloda — APPROVED** [review]

**Review by jbartok — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc (hunk 2 above)
Comment (eskatos) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc:253]: Good point. Yes according to my tests this is required if you upgrade Gradle on a build already imported in the IDE. I added some words in this regard.

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**